src/modules/bulletproofs/main_impl.h

```diff
@@ -0,0 +1,228 @@
+/**********************************************************************
+ * Copyright (c) 2018 Andrew Poelstra                                 *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+
+#ifndef SECP256K1_MODULE_BULLETPROOF_MAIN_IMPL
+#define SECP256K1_MODULE_BULLETPROOF_MAIN_IMPL
+
+#include "group.h"
+#include "scalar.h"
+
+#include "modules/commitment/main_impl.h"
+
+struct secp256k1_bulletproof_generators {
+    size_t n;
+    secp256k1_ge *gens;
+    secp256k1_ge *blinding_gen;
+};
+
+#include "modules/bulletproofs/inner_product_impl.h"
+#include "modules/bulletproofs/rangeproof_impl.h"
+#include "modules/bulletproofs/util.h"
+
+secp256k1_bulletproof_generators *secp256k1_bulletproof_generators_create(const secp256k1_context *ctx, const secp256k1_generator *blinding_gen, size_t n, size_t precomp_n) {
```

Comment:
`s/secp256k1_bulletproof_generators_create/secp256k1_bulletproof_generators_generate`
would be slightly more consistent with the naming in the generators module